benchexec/tablegenerator/statisticstex.py

```diff
@@ -0,0 +1,295 @@
+% Arguments: benchmark name, run-set name, category, status, column name, statistic, value
+\providecommand\StoreBenchExecResult[7]{\expandafter\newcommand\csname#1#2#3#4#5#6\endcsname{#7}}%
+"""
+
+
+class LatexCommand:
+    """Data holder for latex command."""
+
+    def __init__(self, benchmark_name="", display_name=""):
+        self.benchmark_name = LatexCommand.format_command_part(str(benchmark_name))
+        self.display_name = LatexCommand.format_command_part(str(display_name))
+        self.column_title = ""
+        self.column_category = ""
+        self.column_subcategory = ""
+        self.stat_type = ""
+        self.value = None
+
+    def set_command_part(self, part_name: str, part_value) -> "LatexCommand":
+        """Sets the value of the command part
+
+        Available part names:
+            benchmark_name, display_name, column_title, column_category, column_subcategory, stat_type
+
+        Args:
+            part_name: One of the names above
+            part_value: The value to be set for this command part
+
+        Returns:
+            This LatexCommand
+        """
+        self.__dict__[part_name] = LatexCommand.format_command_part(str(part_value))
+        return self
+
+    def set_command_value(self, value: str) -> "LatexCommand":
+        """Sets the value for this command
+
+        The value must be formatted. No checks are made in this method.
+        It will be converted to string.
+
+        Args:
+            value: The new command value
+        Returns:
+            This LatexCommand
+        """
+        if value is None:
+            value = ""
+        self.value = str(value)
+        return self
+
+    def to_latex_raw(self) -> str:
+        """Prints latex command with raw value (e.g. only number, no additional latex command)."""
+        return self._get_command_formatted(self.value)
+
+    def __repr__(self):
+        return "\\StoreBenchExecResult{%s}{%s}{%s}{%s}{%s}{%s}" % (
+            self.benchmark_name,
+            self.display_name,
+            self.column_title,
+            self.column_category,
+            self.column_subcategory,
+            self.stat_type,
+        )
+
+    def _get_command_formatted(self, value: str) -> str:
+        """Formats the command with all parts and appends the value
+
+        To use a custom format for the value, for example
+            \\StoreBenchExecResult{some}{stuff}...{last_name_part}{\\textbf{value}}
+        format the value and give it to this function
+        """
+        if not value:
+            value = ""
+        return str(self) + "{%s}" % value
+
+    @staticmethod
+    def format_command_part(name: str) -> str:
+        name = re.sub(
+            "[0-9]+", lambda match: util.number_to_roman_string(match.group()), name
+        )
+
+        name = re.split("[^a-zA-Z]", name)
+
+        name = "".join(util.cap_first_letter(word) for word in name)
+
+        return name
+
+
+class DuplicationHolder:
```

Comment:
It is weird that `create_duplication_dictionary` returns a data structure that contains a field `used` that is not related at all to `create_duplication_dictionary` and only there because it happens to be useful for the one caller of `create_duplication_dictionary`.

And the field `duplication` seems redundant because these objects will only be used after being stored in a dict under the very same key.

The types of `DuplicationHolder` and `create_duplication_dictionary` force `pre_process` to return a `str`, but this seems unnecessary.

But anyway, I also think this is over-engineered. Inside `write_tex_command_table` there could simply exist two (local) dicts `name_count` and `current_name_index` or similarly. The former can probably be created in a one-liner using [`collections.Counter`](https://docs.python.org/3/library/collections.html#collections.Counter). The latter could be a `defaultdict(int)` or so.